### PR TITLE
Adapt the data-plane-adoption doc/tests to the new GlanceAPI CRD

### PR DESCRIPTION
**Pull request by fmount**
We recently introduced the ability to deploy a list of glanceAPI in the OpenStackControlPlane. This feature is not required in the adoption context, but it's critical to adapt the Glance deployment to the new api.
It might be relevant in the future as long as DCN environments will be adopted.

### CODE REVIEW

**Comment by fmount** [comment]
@jistr @fao89 I'm concerned about the following:

[1] depends-on [2], and [3] depends on [1].

However, we recently introduced a `data-plane-adoption` job in [1] that will fail, preventing [1] to merge before [3].
This is a situation where in order to merge a change in the `openstack-operator` I have to merge this first, otherwise I'm not going to get a green job in [1].
Basically if you look at [4], the fix for the fail in the context of [1] is provided by this patch, and I'm not sure multiple `Depends-On` are going to work in [1].
Any thoughts? I'll follow up on this as long as I have more details.

[1] https://github.com/openstack-k8s-operators/openstack-operator/pull/587
[2] https://github.com/openstack-k8s-operators/glance-operator/pull/384
[3] [this patch](https://github.com/openstack-k8s-operators/data-plane-adoption/pull/221)
[4] [zuul failure](https://logserver.rdoproject.org/87/587/a2c7772ef9b10a4217c1e178162a5737c2359206/github-check/cifmw-data-plane-adoption-OSP-17-to-extracted-crc/71dacac/job-output.txt)

**Comment by cescgina** [comment]
> @jistr @fao89 I'm concerned about the following:
> 
> [1] depends-on [2], and [3] depends on [1].
> 
> However, we recently introduced a `data-plane-adoption` job in [1] that will fail, preventing [1] to merge before [3]. This is a situation where in order to merge a change in the `openstack-operator` I have to merge this first, otherwise I'm not going to get a green job in [1]. Basically if you look at [4], the fix for the fail in the context of [1] is provided by this patch, and I'm not sure multiple `Depends-On` are going to work in [1]. Any thoughts? I'll follow up on this as long as I have more details.
> 
> [1] [openstack-k8s-operators/openstack-operator#587](https://github.com/openstack-k8s-operators/openstack-operator/pull/587) [2] [openstack-k8s-operators/glance-operator#384](https://github.com/openstack-k8s-operators/glance-operator/pull/384) [3] [this patch](https://github.com/openstack-k8s-operators/data-plane-adoption/pull/221) [4] [zuul failure](https://logserver.rdoproject.org/87/587/a2c7772ef9b10a4217c1e178162a5737c2359206/github-check/cifmw-data-plane-adoption-OSP-17-to-extracted-crc/71dacac/job-output.txt)

I'm not 100% it will work, but maybe we should add this a depends-on this change in the openstack-operator patch and use the job running there as voting for this patch. That adoption job should use this PR and the rest of the changes

**Comment by fmount** [comment]
> > @jistr @fao89 I'm concerned about the following:
> > [1] depends-on [2], and [3] depends on [1].
> > However, we recently introduced a `data-plane-adoption` job in [1] that will fail, preventing [1] to merge before [3]. This is a situation where in order to merge a change in the `openstack-operator` I have to merge this first, otherwise I'm not going to get a green job in [1]. Basically if you look at [4], the fix for the fail in the context of [1] is provided by this patch, and I'm not sure multiple `Depends-On` are going to work in [1]. Any thoughts? I'll follow up on this as long as I have more details.
> > [1] [openstack-k8s-operators/openstack-operator#587](https://github.com/openstack-k8s-operators/openstack-operator/pull/587) [2] [openstack-k8s-operators/glance-operator#384](https://github.com/openstack-k8s-operators/glance-operator/pull/384) [3] [this patch](https://github.com/openstack-k8s-operators/data-plane-adoption/pull/221) [4] [zuul failure](https://logserver.rdoproject.org/87/587/a2c7772ef9b10a4217c1e178162a5737c2359206/github-check/cifmw-data-plane-adoption-OSP-17-to-extracted-crc/71dacac/job-output.txt)
> 
> I'm not 100% it will work, but maybe we should add this a depends-on this change in the openstack-operator patch and use the job running there as voting for this patch. That adoption job should use this PR and the rest of the changes

ack, I'll try that after [2] (the glance patch) merges, so we have one less dependency to worry about.
I was hoping to reach a level where I could see everything green before starting the merge process, but that seems too hard. Right now I can't add the `Depends-On` to the openstack-operator patch because there's already a depends-on and I'm not sure it works w/ >1 dependency.

**Review by cescgina — COMMENTED** [review]
lgtm, we need to wait until https://github.com/openstack-k8s-operators/openstack-operator/pull/587 merges to get this in, also adoption job running in that PR uses this one as depends-on and is working:

**Comment by fmount** [comment]
As per the logs glance seems happy, but it fails due to:

```
TASK [dataplane_adoption : force latest ansible-runner image] ******************
fatal: [localhost]: FAILED! => {"changed": true, "cmd": "set -euxo pipefail\n\n\noc login -u kubeadmin -p 123456789\n\n# openstack-operator catalog pins the sha256, which might differ from latest with time\noc patch csv -n openstack-operators openstack-ansibleee-operator.v0.0.1  --type='json' -p='[{\n  \"op\":\"replace\", \"path\":\"/spec/install/spec/deployments/0/spec/template/spec/containers/1/env/0\",\n  \"value\": {\"name\": \"RELATED_IMAGE_ANSIBLEEE_IMAGE_URL_DEFAULT\", \"value\": \"quay.io/openstack-k8s-operators/openstack-ansibleee-runner:latest\"}}]'\n", "delta": "0:00:00.261982", "end": "2023-12-15 12:22:39.497558", "msg": "non-zero return code", "rc": 1, "start": "2023-12-15 12:22:39.235576", "stderr": "+ oc login -u kubeadmin -p 123456789\nError from server (InternalError): Internal error occurred: unexpected response: 400", "stderr_lines": ["+ oc login -u kubeadmin -p 123456789", "Error from server (InternalError): Internal error occurred: unexpected response: 400"], "stdout": "WARNING: Using insecure TLS client config. Setting this option is not supported!", "stdout_lines": ["WARNING: Using insecure TLS client config. Setting this option is not supported!"]}
```

**Comment by fmount** [comment]
recheck

**Review by gibizer — APPROVED** [review]
looks good

**Review by jistr — APPROVED** [review]

**Review by marios — APPROVED** [review]
lgtm going to merge it has enough approvals and there is a patch queue behind this :)

job was green in friday run

**Pull request merged**
